### PR TITLE
Able to view a file.

### PR DESCRIPTION
**Pull request by caseylargent**
Addresses #221. Need some suggestions for how to delete the directory. I'm thinking about just using the shell to delete the directory when the app closes.

### CODE REVIEW

**Comment by djw8605** [comment]
There's gotta be a way to use ng-disabled instead of setting a CSS.

**Comment by caseylargent** [comment]
Do you think we should change those css modifications to be booleans that are set instead?

**Comment by caseylargent** [comment]
so that we can use ng-disabled?

File: HCCGo/app/js/fileManageService.js [hunk 1]
```diff
@@ -230,13 +232,40 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
+                            shell.moveItemToTrash(path);
+                        }, 3000);
+                    });
+                });
+            }
+        });
+    };
+
+   service.isDirectory = function(id) {
+       angular.element("#btnView").attr('disabled', '');
```
Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: I think this should be removed (along with the other css attr changes).  Instead, ng-disabled in the HTML should be used.  I'm not sure what element in the `focusFile` object would say it's a directory.

File: HCCGo/app/js/fileManageService.js [hunk 2]
```diff
@@ -175,68 +181,69 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
 	   
        localRead(service.localWD);
    };
    
    service.verifyUpload = function () {
-	  angular.element('#btnUpload').attr('disabled', '');
-	  service.userUpAuth = true;
-	  service.processStatus = true;
+        service.canUpload = false;
+	    service.userUpAuth = true;
```
Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: Why all this new weird space.

File: HCCGo/app/js/fileManageService.js [hunk 3]
```diff
@@ -175,68 +181,69 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
+        connectionService.localSize(String(service.localWD + "/" + service.localFocus)).then( function(ldata) {
+            if (service.remoteWD.indexOf(service.workWD) > -1) {
+                connectionService.runCommand("lfs quota -g `id -g` /work").then(function(data) {
+                    service.processStatus = false;
+                    service.accuSize = ldata;
+
+                    let reported_output = data.split("\n")[2];
+                    let split_output = $.trim(reported_output).split(/[ ]+/);
+                    service.diskAvail = Math.floor(((split_output[3] - split_output[1]) / split_output[3])*100);
+                    service.diskQuota = Math.floor(((ldata / Math.pow(1024, 1)) / split_output[3])*100);
```
Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: Weird spaces!

File: HCCGo/app/js/fileManageService.js [hunk 4]
```diff
@@ -175,68 +181,69 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
+        connectionService.runCommand("du -sb " + String(service.remoteWD + "/" + service.remoteFocus)).then(function (data) {
+            service.processStatus = false;
+            let data_response = data.split(/[	]+/); //NOTE: Matches tab spaces
+            service.accuSize = data_response[0];
+            disk.check(service.localWD, function(err, info) {
+                service.diskQuota = Math.floor((data_response[0]/info.available)*100);
+                service.diskAvail = Math.floor((info.free/info.total)*100);
+            });
+        });
    };
```
Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: More weird spaces!

File: HCCGo/app/js/fileManageService.js [hunk 5]
```diff
@@ -175,68 +181,69 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
    
    service.verifyDownloadCancel = function () {
-	  service.userDownAuth = false;
-      notifierService.warning('Action cancelled by user.');
+        service.userDownAuth = false;
+        notifierService.warning('Action cancelled by user.');
```
Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: Spaces!

**Comment by djw8605** [comment]
Nope, still lots of spacing issues :)

**Comment by caseylargent** [comment]
Ugh... I went through the entire file and did the spacing... does it really matter that much?

**Comment by djw8605** [comment]
As long as everything still works, download, upload, and view.  But if anything breaks, git blame is going to blame you rather than Jerrod :)

-Derek

**Comment by djw8605** [comment]
Oh, and 1 more thing.  It needs some sort of progress bar.  Downloading / opening...  It needs some feedback to the user when you click on "view".  If it's a larger file, it may take a few seconds for anything to show up.

**Comment by caseylargent** [comment]
I don't know if we would necessarily need a progress bar for the dowloading/opening, but instead should be some spinning image in my opinion. Mostly because the progress bar would move decently fast fi all the logic for increasing the bar is placed within the file management js file

**Comment by djw8605** [comment]
I like a progress bar because it actually shows the progress of the download.  You can do whatever, but it needs some sort of feedback.

**Comment by djw8605** [comment]
I like the rotating circle.  Though, I'm worried about the timing of the file removal.  I tried opening a file with a `.slurm` ending, which obviously didn't have a file association.  By the time I selected the textedit application to open, it had already been removed and textedit failed.

I think the `tmp` package will automatically delete the file.  Lets let it do the deleting.

File: HCCGo/app/js/fileManageService.js [hunk 6]
```diff
@@ -173,241 +183,272 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
...
-      let boolStarter = true;
-	  service.processStatus = true;
-	  service.userDownAuth = false;
-
-      // Runs file upload
-      connectionService.downloadFile(String(service.localWD + "/"), 
+   service.viewFile = function() {
+        service.viewing = true;
+        connectionService.getFileSize(service.focus.location).then(function(size){
+            if (size > 5*105*1024) {
```
Comment (djw8605) [inline HCCGo/app/js/fileManageService.js]: I think this should be 5*1024*1024 (5MB)

**Pull request merged**